### ISSUE TITLE
Suggestion: Add a .clang-format file

### ISSUE THREAD

**Issue by lmichaelis** [Try/OpenGothic#304]
While working on #271, I've missed being able to automatically format code to the point that I sometimes just didn't do it. Having a `.clang-format` file [like the one I have in phoenix](https://github.com/lmichaelis/phoenix/blob/main/.clang-format) would make contributing easier and get rid of any ambiguity concerning formatting. You can even test out all the parameters it provides in [an online tool](https://zed0.co.uk/clang-format-configurator/) :)

**Comment by lmichaelis** [comment]
Yeah I think that's not possible. But changing that part of the code style shouldn't be too bad since it mostly just affects lines with closing braces, right? I'm not aware of too many cases where block closing braces are on the same line as some other code.

**Comment by Reveares** [comment]
I also think that it would be useful to add a `.clang-format`, just because CLion does not handle the current style so well.

> And changing code-style to else, mean one huge commit that will destroy git-blame :/

This problem is relatively easy to fix, this article describes how https://www.stefanjudis.com/today-i-learned/how-to-exclude-commits-from-git-blame/

**Comment by Try** [comment]
Good point, thanks @Reveares !

There is one more thing I'm wondering about:
Maybe there a way to instruct `clang-format` to format now all things, but few selected ones? Reason for that: I'm not ready drop habit of  unusual code-style for brackets

**Comment by CReimer** [comment]
+1 for switching to a more common code style with proper support in IDEs. C++ is difficult enough for me right now. Having to fight against CLion every time I hit Enter is painful.

**Comment by lmichaelis** [comment]
> Good point, thanks @Reveares !
> 
> There is one more thing I'm wondering about: Maybe there a way to instruct `clang-format` to format now all things, but few selected ones? Reason for that: I'm not ready drop habit of unusual code-style for brackets

@Try I just found this in the clang-format docs: It is possible to use [a Git plugin](https://clang.llvm.org/docs/ClangFormat.html#git-integration) which only touches lines changed in Git. Visual Studio probably has a setting to that effect as well.

You could also just [manually run it](https://clang.llvm.org/docs/ClangFormat.html#standalone-tool) on files you're comfortable with.

**Comment by CReimer** [comment]
I found out @Try s brace style is called Ratliff style. The problem is: Nobody wants that brace style.
There is only one tool which supports this brace style. It's called astyle. But it doesnt really work for me either. A lot of other stuff is not supported or just ignored

I tried to set it with CLion and I experimented with clang-format. They both don't support Ratliff style.

Or to summarize: It's an extremely exotic code style that no one wants to support.

But let's set it aside for the moment. There's still a lot of other inconsistencies e.g.

Reference alignment and Pointer alignment: Sometimes left, sometimes right

Or commas. Sometimes there's a space after a comma (as it should be) and sometimes there's none.

**Comment by PoetaKodu** [comment]
I'd also really appreciate using a more common formatting style. I found it very confusing personally.

**Comment by PawelBis** [comment]
I fear no code. But that style... it scares me.

Seriously, I love this project. However I think that the current code style is just killing the appeal of it for a lot of potential contributors.
